libs/wingsdk/src/simulator/simulator.ts

```diff
@@ -643,30 +681,19 @@ export class Simulator {
    */
   private tryResolveTokens(obj: any): any {
     if (typeof obj === "string") {
-      if (isToken(obj)) {
-        const ref = obj.slice(2, -1);
-        const [path, rest] = ref.split("#");
-        const config = this.getResourceConfig(path);
-        if (rest.startsWith("attrs.")) {
-          const attrName = rest.slice(6);
-          const attr = config?.attrs[attrName];
-
-          // we couldn't find the attribute. this doesn't mean it doesn't exist, it's just likely
-          // that this resource haven't been started yet. so return `undefined`, which will cause
-          // this resource to go back to the init queue.
-          if (!attr) {
+      let resolveCount = 0; // prevent infinite loop, just in case
+      while (resolveCount < 100) {
+        resolveCount += 1;
+        const matches = obj.match(SIMULATOR_TOKEN_REGEX);
+        if (!matches) {
+          break;
+        }
+        for (const match of matches) {
+          const resolved = this.tryResolveToken(match);
+          if (resolved === undefined) {
```

Comment:
what if a property is set to wing's `nil`?

Reply:
good point, thanks for catching this. I think I was assuming the token always resolves to a string, but we don't really know that for sure, so I should probably model this a bit differently. Working on a fix